File: Week_04/55/jump_game.go
package main

func canJump(nums []int) bool {
	n := len(nums)
	rightMost := 0

	for i := 0; i < n; i++ {
		if i <= rightMost {
			rightMost = max(rightMost, i+nums[i])
		}
		if rightMost >= n-1 {
			return true
		}
	}
	return false
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

File: Week_04/55/jump_game.py
from typing import List


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        right_most = 0
        n = len(nums)
        for i in range(0, n):
            if i <= right_most:
                right_most = max(i + nums[i], right_most)
            if right_most >= n - 1:
                return True
        return False
